pkg/handlers/s3_handler.go

```diff
@@ -157,10 +158,24 @@ func (h *s3Handler) OnAdd(ns string, vc kvcv1.VolumeConfig, controllerRef metav1
 	for _, kvcName := range kvcNames {
 		err := waitForPodSuccess(podClient, kvcName, ns, timeout)
 		if err != nil {
+			req := h.k8sClientset.CoreV1().RESTClient().Get().Namespace(ns).Name(kvcName).Resource("pods").SubResource("log")
```

Comment:
Me and @ashahba had discussed using the RESTClient as one possibility and the dynamic client as the other for getting the logs. I think using the dynamic client would be the best way to do it since it would keep the operations consistent. But the problem is that getting the logs is only applicable for pods, so including a method in genericClient would be not so ideal. Maybe a helper method, which takes in the dynamic client and returns the log would be useful? In that way it does not live in genericClient, but still would use it. WDYT @balajismaniam ?

Reply:
I like the helper method route. May be we can explore that.